src/Core/Domain/Shop/ValueObject/ShopConstraint.php

```diff
@@ -164,4 +164,19 @@ public function isEqual(self $constraint): bool
 
         return false;
     }
+
+    public function isSingleShopContext(): bool
+    {
+        return null !== $this->shopId;
+    }
+
+    public function isGroupShopContext(): bool
```

Comment:
```suggestion
    public function isShopGroupContext(): bool
```
To be consistent with `$this->shopGroupId: ShopGroupId`